File: solutions/c/2.c
#include <stdio.h>
#include <stdlib.h>
#include <inttypes.h>

#define MAX_SEQUENCE 4000000

int main(void)
{
  // Fibonnaci sequence actually starts 1, 2 but pretending it is 1, 1, 2 allows
  // us to use the same logic for every step instead of treating the first one
  // as a special case
  uint32_t sequence[] = {1, 1, 2};
  uint64_t sum = 0;

  while (sequence[2] <= MAX_SEQUENCE)
  {
    if (sequence[2] % 2 == 0)
    {
      sum += sequence[2];
    }

    // Move all elements back one space, then recalculate current element
    sequence[0] = sequence[1];
    sequence[1] = sequence[2];
    sequence[2] = sequence[0] + sequence[1];
  }

  printf("%" PRIu64 "\n", sum);

  return EXIT_SUCCESS;
}

File: solutions/python/2.py
max_number = 4000000
fib_sequence = [1, 1, 2]
total = 0

while fib_sequence[2] <= max_number:
  if fib_sequence[2] % 2 == 0:
      total += fib_sequence[2]

  # Move all elements back one place and recalculate the 3rd
  fib_sequence[0] = fib_sequence[1]
  fib_sequence[1] = fib_sequence[2]
  fib_sequence[2] = fib_sequence[0] + fib_sequence[1]

print("Total of all even Fibonnaci numbers up to " + str(max_number) + " is: " + str(total))
